src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -269,7 +269,8 @@ private void ConsumeFromRemainingBuffer(int bytesToConsume)
             else
             {
                 Debug.Assert(_pool.UsingProxy);
-                await WriteAsciiStringAsync(uri.IdnHost).ConfigureAwait(false);
```

Comment:
What about the if block above? Don't we need to fix that case as well?

Reply:
Yes, we need to fix that as well.  The fix there is more involved because it needs to happen as part of the construction of the connection key -- basically we should detect that the hostname is an IPv6 address when constructing the key, and construct the hostname using [].

You won't be able to test this using a proxy as you're doing here.  But you should at least be able to test against IPv6Loopback.  There's already a test for IP address based Uris -- GetAsync_IPBasedUri_Success.  Unfortunately it's not validating the Host header.  It should be modified to validate the Host header.  And it would be nice to add https cases to that test as well.

Reply:
Thanks for pointing this out! I will make a fix and add tests to it.

Reply:
Fix the above `if` by changing `GetConnectionKey` method.

We still need to special case this `else` statement [for `Proxy` case](https://github.com/dotnet/corefx/blob/50d4e1ce620f26ec4aebe26c121a769fc8e724ab/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs#L180), because for this case, we pass `null` as host, thus `_pool.HostHeaderValueBytes == null`, and here is the only place to append `[]`